src/gpuarray_buffer_cuda.c

```diff
@@ -71,9 +71,13 @@ cuda_context *cuda_make_ctx(CUcontext ctx, int flags) {
   if (err != CUDA_SUCCESS) {
     goto fail_stream;
   }
-  err = cuStreamCreate(&res->mem_s, CU_STREAM_NON_BLOCKING);
-  if (err != CUDA_SUCCESS) {
-    goto fail_mem_stream;
+  if (ISSET(res->flags, GA_CTX_SINGLE_STREAM)) {
+    res->mem_s = res->s;
+  } else {
+    err = cuStreamCreate(&res->mem_s, 0);
```

Comment:
Yes, this will lead to less problems for interacting with libraries that use the NULL stream since all the internal streams will synchronize with it implicitly.